ios/BlindThresholdBls.m

```diff
@@ -1,14 +1,119 @@
 #import "BlindThresholdBls.h"
+#import "Headers/threshold.h"
+#import <React/RCTLog.h>
+#import <Foundation/Foundation.h>
+#import <stdlib.h>
+#import <stdint.h>
 
-
-@implementation BlindThresholdBls
+@implementation BlindThresholdBls {
+  Token_PrivateKey* blindingFactor;
+  Buffer messageBuf;
+}
 
 RCT_EXPORT_MODULE()
 
-RCT_EXPORT_METHOD(sampleMethod:(NSString *)stringArgument numberParameter:(nonnull NSNumber *)numberArgument callback:(RCTResponseSenderBlock)callback)
+RCT_REMAP_METHOD(blindMessage,
+                 message:(NSString *) message
+                 resolver:(RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject)
 {
-    // TODO: Implement some actually useful functionality
-    callback(@[[NSString stringWithFormat: @"numberArgument: %@ stringArgument: %@", numberArgument, stringArgument]]);
+  @try {
+    RCTLogInfo(@"Preparing blind message buffers");
+    NSData *messageData = [message dataUsingEncoding:NSUTF8StringEncoding];
+    messageBuf.ptr = (uint8_t*)[messageData bytes];
+    messageBuf.len = [messageData length];
+//    uint8_t * messageBytes = malloc(length);
+//    memcpy(messageBytes, [messageData bytes], length);
+    Buffer blindedMessageBuf;
+    
+    RCTLogInfo(@"Preparing blinding seed");
+    NSMutableData* seedData = [NSMutableData dataWithCapacity:32];
```

Comment:
`SecRandomCopyBytes` seems slightly easier https://stackoverflow.com/a/34602643 - though i would use `kSecRandomDefault` instead of null https://developer.apple.com/documentation/security/1399291-secrandomcopybytes